### ISSUE TITLE
Mesh picking - transparency on background issue

### ISSUE THREAD

**Issue by S41L0R** [KillzXGaming/Cafe-Shader-Studio#17]
Transparency doesn't fully work when rendering onto background when object is selected with mesh picking. This problem seems to come from Picking.frag or how it is used.

Tested in Gagb_MarioCircuit in MK8 WiiU

![image](https://user-images.githubusercontent.com/60116462/138556651-3f590ad7-6020-4b9b-b75e-88fe44eff53c.png)

**Comment by S41L0R** [comment]
I agree it's not a big deal.. but the problem here is that it's only rendering if it's on top of the background.

**Comment by KillzXGaming** [comment]
An idea which will probably be used. The current method draws a solid mesh and offsets it creating an outline but a better way of handling it would be outputting a second color pass that has color or not depending on what is selected. Then it can be created later during the post processing. http://www.geoffprewett.com/blog/software/opengl-outline/index.html

This method will support proper outlines for transparent models and also will save on performance not needing to be redrawn twice. Also with this I can add a proper selection highlight color that adds to the whole model.

The tricky part is implementing it with custom shaders. I'll need to try adding onto shaders during compilation with custom editor code.
